test/test_jit.py

```diff
@@ -3959,6 +3959,91 @@ def test_while(a, b):
                 return a + b
             ''')
 
+    def test_optional_refinement(self):
+        with self.disableModuleHook():  # TODO:
```

Comment:
Hmm. This isn't schematizable, so it would need special logic. I don't think this is an actual use case - I grep'd around for "Optional[Optional[" and didn't find anything. I think erroring and waiting to see if it ever comes up would be sufficient.